### PR TITLE
Homework solution

### PR DESCRIPTION
**Pull request by timmygee**

### CODE REVIEW

**Comment by timmygee** [comment]
Pull request currently missing "On the project list view, show how much time was spent per project in total and how much time the current user spent" bonus. Working on that now

File: timetracker/entries/templates/projects.html [hunk 1]
```diff
@@ -12,8 +12,17 @@
   <ul class="list-group col-md-3">
     {% for project in project_list %}
       <li class="list-group-item">
-        <a href="{% url 'project-detail' pk=project.pk %}">{{ project.name }}</a> (<a href="{% url 'client-detail' pk=project.client.pk %}">{{ project.client.name }}</a>)
-        <a class="btn btn-warning btn-xs pull-right" href="{% url 'project-detail' pk=project.pk %}">Edit</a>
+        {% if project.author == request.user %}<a href="{% url 'project-detail' pk=project.pk %}">{% endif %}
```
Comment (timmygee) [inline timetracker/entries/templates/projects.html:15]: Is there a way to omit the <a> and </a> tags conditionally that I just don't know about? That is, without writing a custom template tag / filter?

File: timetracker/entries/migrations/0004_auto_20150910_0348.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('entries', '0003_create_fk_to_client'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='client',
+            name='author',
+            field=models.ForeignKey(to=settings.AUTH_USER_MODEL, default=None),
```
Comment (MarkusH) [inline timetracker/entries/migrations/0004_auto_20150910_0348.py]: This won't work with existing rows

File: timetracker/entries/templates/projects.html (hunk 1 above)
Comment (MarkusH) [inline timetracker/entries/templates/projects.html:15]: I normally go with

```
{% if %}<a>text</a>{% else %}text{% endif %}
```

File: timetracker/entries/migrations/0004_auto_20150910_0348.py (hunk 2 above)
Comment (timmygee) [inline timetracker/entries/migrations/0004_auto_20150910_0348.py]: What is the solution there? Set the field to have null=True in models.py?

File: timetracker/entries/templates/projects.html (hunk 1 above)
Comment (timmygee) [inline timetracker/entries/templates/projects.html:15]: Ok. In this case we have 2 unique links on one line that can be "on on", "on off" or "off off" so there's going to be a bit of repetition. Ok to leave as it is or should we use your approach?

File: timetracker/entries/migrations/0004_auto_20150910_0348.py (hunk 2 above)
Comment (MarkusH) [inline timetracker/entries/migrations/0004_auto_20150910_0348.py]: 1. Add column with `null=True`
2. Propagate value for existing rows
3. Drop null constraint

If you don't have a solution for 2, and existing entries may not be NULL you have a problem you can't solve.

Comment (MarkusH) [inline timetracker/entries/migrations/0004_auto_20150910_0348.py]: E.g. fall back to first superuser would be a possible fallback, but I don't like it.

File: timetracker/entries/templates/projects.html (hunk 1 above)
Comment (MarkusH) [inline timetracker/entries/templates/projects.html:15]: I don't mind the current implementation.

**Pull request closed without merging**